document-generator-prosecution/src/test/java/uk/gov/companieshouse/document/generator/prosecution/ProsecutionDocumentInfoServiceTest.java

```diff
@@ -1,101 +1,91 @@
-                    + "then that response is returned.")
     @Test
-    void testGetDocumentInfoForUltimatumNoProsecutionCaseFound()
-                    throws SdkException {
-        ProsecutionDocumentInfoService prosDocInfoService = new ProsecutionDocumentInfoService(
-                        docInfoBuilderProvider, prosecutionClient);
-        when(prosecutionClient.getProsecutionCase(PROSECUTION_CASE_URI, REQUEST_ID))
-                        .thenThrow(SdkException.class);
-
-        DocumentInfoRequest docInfoReq = new DocumentInfoRequest();
-        docInfoReq.setResourceUri(ULTIMATUM_RESOURCE_URI);
-        docInfoReq.setRequestId(REQUEST_ID);
-        assertThrows(DocumentInfoException.class,
-                        () -> prosDocInfoService.getDocumentInfo(docInfoReq));
+    @DisplayName("Test using a prosecution case that has an ULTIMATUM_ISSUED status "
+            + "should call the handler for SJPn")
+    public void testGetDocumentInfoWithUltimatumIssuedCase() throws Exception {
+        ProsecutionCase prosecutionCase = new ProsecutionCase();
+        prosecutionCase.setStatus(ProsecutionCaseStatusApi.ULTIMATUM_ISSUED);
+        DocumentInfoRequest request = getDocumentInfoRequest(resourceUri, requestId);
+        ProsecutionDocument document = new ProsecutionDocument();
+        document.setProsecutionCase(prosecutionCase);
+        when(mockHandler.getProsecutionDocument(resourceUri)).thenReturn(document);
+        DocumentInfoResponse response = new DocumentInfoResponse();
+        when(mockHandler.getDocumentResponse(document, requestId, prosecutionCase.getStatus())).thenReturn(response);
+        prosecutionDocumentInfoService.getDocumentInfo(request);
+        verify(mockHandler, times(1)).getProsecutionDocument(resourceUri);
+        verify(mockHandler, times(1)).getDocumentResponse(document, requestId, prosecutionCase.getStatus());
     }
 
-    @DisplayName("Given a request to produce an ultimatum for a given prosecution case, "
-                    + "when the UltimatumDocumentInfoService successfully produces a DocumentInfoResponse,"
-                    + "then that response is returned.")
     @Test
-    void testGetDocumentInfoForUltimatumErrorBuildingDocInfo()
-                    throws SdkException, DocumentInfoCreationException {
-        ProsecutionDocumentInfoService prosDocInfoService = new ProsecutionDocumentInfoService(
-                        docInfoBuilderProvider, prosecutionClient);
-        when(prosecutionClient.getProsecutionCase(PROSECUTION_CASE_URI, REQUEST_ID))
-        .thenReturn(prosecutionCase);
-        when(docInfoBuilderProvider.builder()).thenReturn(docInfoBuilder);
-        when(docInfoBuilder.build())
-        .thenThrow(DocumentInfoCreationException.class);
+    @DisplayName("Test using a prosecution case that has a REFERRED status should throw "
+            + "DocumentInfoException")
```

Comment:
This is a bit misleading. Some people use the unit tests to understand the code. The document info service doesn't check the status of the document. I would keep the tests generic:

1. Test successful retrieval of document info
2. Test handler exception is correctly handled when getting document data
3. Test handler exception is correctly handled when creating document response

Reply:
As the DocumentInfoService has been refactored, the unit tests have also been refactored to test the successful generation of a document as well as if a handler exception was returned by the handler.